README.rst
..
    This file is part of CERN Open Data Portal.
    Copyright (C) 2017 CERN.

    CERN Open Data Portal is free software; you can redistribute it
    and/or modify it under the terms of the GNU General Public License as
    published by the Free Software Foundation; either version 2 of the
    License, or (at your option) any later version.

    CERN Open Data Portal is distributed in the hope that it will be
    useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CERN Open Data Portal; if not, write to the
    Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
    MA 02111-1307, USA.

    In applying this license, CERN does not
    waive the privileges and immunities granted to it by virtue of its status
    as an Intergovernmental Organization or submit itself to any jurisdiction.

====================
 cernopendata-theme
====================

.. image:: https://img.shields.io/travis/cernopendata/cernopendata-theme.svg
        :target: https://travis-ci.org/cernopendata/cernopendata-theme

.. image:: https://img.shields.io/coveralls/cernopendata/cernopendata-theme.svg
        :target: https://coveralls.io/r/cernopendata/cernopendata-theme

.. image:: https://img.shields.io/github/tag/cernopendata/cernopendata-theme.svg
        :target: https://github.com/cernopendata/cernopendata-theme/releases

.. image:: https://img.shields.io/pypi/dm/cernopendata-theme.svg
        :target: https://pypi.python.org/pypi/cernopendata-theme

.. image:: https://img.shields.io/github/license/cernopendata/cernopendata-theme.svg
        :target: https://github.com/cernopendata/cernopendata-theme/blob/master/LICENSE


Theme for CERN Open Data Portal.

* Free software: GPLv2 license


CHANGES.rst
..
    This file is part of CERN Open Data Portal.
    Copyright (C) 2017 CERN.

    CERN Open Data Portal is free software; you can redistribute it
    and/or modify it under the terms of the GNU General Public License as
    published by the Free Software Foundation; either version 2 of the
    License, or (at your option) any later version.

    CERN Open Data Portal is distributed in the hope that it will be
    useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
    General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CERN Open Data Portal; if not, write to the
    Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
    MA 02111-1307, USA.

    In applying this license, CERN does not
    waive the privileges and immunities granted to it by virtue of its status
    as an Intergovernmental Organization or submit itself to any jurisdiction.


Changes
=======

Version 1.0.1 (released 2017-01-20)

- Adds missing frontpage.scss and logos.

Version 1.0.0 (released 2017-01-19)

- Initial public release.


run-tests.sh
#!/usr/bin/env sh
# -*- coding: utf-8 -*-
#
# This file is part of CERN Open Data Portal.
# Copyright (C) 2017 CERN.
#
# CERN Open Data Portal is free software; you can redistribute it
# and/or modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation; either version 2 of the
# License, or (at your option) any later version.
#
# CERN Open Data Portal is distributed in the hope that it will be
# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with CERN Open Data Portal; if not, write to the
# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
# MA 02111-1307, USA.
#
# In applying this license, CERN does not
# waive the privileges and immunities granted to it by virtue of its status
# as an Intergovernmental Organization or submit itself to any jurisdiction.


pydocstyle cernopendata_theme tests && \
isort -rc -c -df && \
check-manifest --ignore ".travis-*" && \
python setup.py test


tests/conftest.py
# -*- coding: utf-8 -*-
#
# This file is part of CERN Open Data Portal.
# Copyright (C) 2017 CERN.
#
# CERN Open Data Portal is free software; you can redistribute it
# and/or modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation; either version 2 of the
# License, or (at your option) any later version.
#
# CERN Open Data Portal is distributed in the hope that it will be
# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with CERN Open Data Portal; if not, write to the
# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
# MA 02111-1307, USA.
#
# In applying this license, CERN does not
# waive the privileges and immunities granted to it by virtue of its status
# as an Intergovernmental Organization or submit itself to any jurisdiction.


"""Pytest configuration."""

from __future__ import absolute_import, print_function

import os
import shutil
import tempfile

import pytest
from flask import Flask, render_template
from flask.cli import ScriptInfo
from invenio_assets import InvenioAssets
from invenio_i18n import InvenioI18N
from invenio_theme import InvenioTheme

from cernopendata_theme import config
from cernopendata_theme.views import blueprint


@pytest.yield_fixture()
def instance_path():
    """Instance path."""
    instance_path = tempfile.mkdtemp()

    yield instance_path

    shutil.rmtree(instance_path)


@pytest.yield_fixture()
def static_folder(instance_path):
    """Static file directory."""
    pth = os.path.join(instance_path, 'static')
    if not os.path.exists(pth):
        os.makedirs(pth)
    yield pth
    shutil.rmtree(pth)


@pytest.fixture()
def app(instance_path, static_folder):
    """Flask application fixture."""
    app = Flask(
        'testapp',
        instance_path=instance_path,
        static_folder=static_folder,
    )
    app.config.from_object(config)
    app.config.update(
        TESTING=True
    )
    InvenioTheme(app)
    InvenioI18N(app)

    @app.route('/')
    def index():
        """Test home page."""
        return render_template('cernopendata_theme/page.html')

    app.register_blueprint(blueprint)
    return app


@pytest.yield_fixture()
def script_info(app):
    """Get ScriptInfo object for testing CLI."""
    InvenioAssets(app)
    yield ScriptInfo(create_app=lambda info: app)


tests/test_cernopendata_theme.py
# -*- coding: utf-8 -*-
#
# This file is part of CERN Open Data Portal.
# Copyright (C) 2017 CERN.
#
# CERN Open Data Portal is free software; you can redistribute it
# and/or modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation; either version 2 of the
# License, or (at your option) any later version.
#
# CERN Open Data Portal is distributed in the hope that it will be
# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with CERN Open Data Portal; if not, write to the
# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
# MA 02111-1307, USA.
#
# In applying this license, CERN does not
# waive the privileges and immunities granted to it by virtue of its status
# as an Intergovernmental Organization or submit itself to any jurisdiction.

"""Module tests."""

from __future__ import absolute_import, print_function

import os
import subprocess

from click.testing import CliRunner
from flask_assets import assets

from invenio_assets.cli import collect, npm


def test_version():
    """Test version import."""
    from cernopendata_theme import __version__
    assert __version__


def test_view(app, script_info):
    """Test assets command in assets CLI."""
    static_root = app.extensions['collect'].static_root
    runner = CliRunner()

    # Generate package.json.
    result = runner.invoke(npm, obj=script_info)
    assert result.exit_code == 0

    filepath = os.path.join(app.static_folder, 'package.json')
    assert os.path.exists(filepath)

    current_dir = os.getcwd()
    os.chdir(app.static_folder)
    exit_status = subprocess.call('npm install', shell=True)
    assert exit_status == 0
    os.chdir(current_dir)

    # Run collect
    result = runner.invoke(collect, ['-v'], obj=script_info)
    assert result.exit_code == 0

    # Run build
    result = runner.invoke(assets, ['build'], obj=script_info)
    assert result.exit_code == 0

    with app.test_client() as client:
        response = client.get('/')
        assert response.status_code == 200


cernopendata_theme/version.py
# -*- coding: utf-8 -*-
#
# This file is part of CERN Open Data Portal.
# Copyright (C) 2017 CERN.
#
# CERN Open Data Portal is free software; you can redistribute it
# and/or modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation; either version 2 of the
# License, or (at your option) any later version.
#
# CERN Open Data Portal is distributed in the hope that it will be
# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with CERN Open Data Portal; if not, write to the
# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
# MA 02111-1307, USA.
#
# In applying this license, CERN does not
# waive the privileges and immunities granted to it by virtue of its status
# as an Intergovernmental Organization or submit itself to any jurisdiction.

"""Version information for cernopendata-theme.

This file is imported by ``cernopendata_theme.__init__``,
and parsed by ``setup.py``.
"""

from __future__ import absolute_import, print_function

__version__ = "1.0.1"


cernopendata_theme/__init__.py
# -*- coding: utf-8 -*-
#
# This file is part of CERN Open Data Portal.
# Copyright (C) 2017 CERN.
#
# CERN Open Data Portal is free software; you can redistribute it
# and/or modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation; either version 2 of the
# License, or (at your option) any later version.
#
# CERN Open Data Portal is distributed in the hope that it will be
# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
# General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with CERN Open Data Portal; if not, write to the
# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
# MA 02111-1307, USA.
#
# In applying this license, CERN does not
# waive the privileges and immunities granted to it by virtue of its status
# as an Intergovernmental Organization or submit itself to any jurisdiction.

"""Theme for CERN Open Data Portal."""

from __future__ import absolute_import, print_function

from .version import __version__

__all__ = ('__version__', )
